src/controllers/DefaultController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Reroute plugin for Craft CMS 3.x
+ *
+ * Manage 301/302 redirects in the control panel.
+ *
+ * @link      https://www.trevor-davis.com
+ * @copyright Copyright (c) 2018 Trevor Davis
+ */
+
+namespace davist11\reroute\controllers;
+
+use davist11\reroute\Reroute;
+
+use Craft;
+use craft\web\Controller;
+use davist11\reroute\models\RerouteModel;
+
+/**
+ * @author    Trevor Davis
+ * @package   Reroute
+ * @since     2.0.0
+ */
+class DefaultController extends Controller
+{
+    // Public Methods
+    // =========================================================================
+
+    /**
+     * @return mixed
+     */
+    public function actionSave()
+    {
+        $this->requirePostRequest();
+
+        // Is this an existing entry or a new entry?
+        if ($id = Craft::$app->request->post('rerouteId')) {
+            $model = Reroute::getInstance()->rerouteService->getById($id);
+        } else {
+            $model = new RerouteModel();
+        }
+
+        // Get the submitted data
+        $data = Craft::$app->request->post('reroute');
+        $model->oldUrl = $data['oldUrl'];
+        $model->newUrl = $data['newUrl'];
+        $model->method = $data['method'];
+
+        $success = false;
+
+        if($model->validate()) {
+            $success = Reroute::getInstance()->rerouteService->save($model);
```

Comment:
I thiiink you can also do `Reroute::$plugin->rerouteService->save($model);` instead of `getInstance()`. Not sure it matters which way you do it though.